### ISSUE TITLE
Video request returns 404/403

### ISSUE THREAD

**Issue by anast3t** [thelevicole/youtube-to-html5-loader#13]
When loading player, there is extremly high chance to get 404 or 403 on responce (even on example page). As temporary solution i just filter them in loop, but maybe it can be fixed.
![image](https://user-images.githubusercontent.com/26357306/118760113-6f7ffd80-b89c-11eb-8dbf-4d42fb26e13a.png)

(code example, maybe i'm just doing something wrong)
```
let gotFlag = false
while (!gotFlag) {
      console.log("trying...")
      let player = new yt2html({
	      withAudio: true,
	      autoload: false,
      })
      player.addFilter('video.source', function (source, data, element, format, streams) {
	      gotFlag = true
	      if (Array.isArray(streams) && streams.length > 1) {
		      const options = streams;
		      options.sort(function (a, b) {
			      const aLabel = parseInt(a.label);
			      const bLabel = parseInt(b.label);
      
			      if (aLabel < bLabel) {
				      return -1;
			      }
      
			      if (aLabel > bLabel) {
				      return 1;
			      }
      
			      return 0;
		      });
      
		      for (let index = 0; index < options.length; index++) {
			      const stream = options[index];
			      const option = document.createElement('option');
			      option.value = stream.url;
			      option.text = `${stream.label}`;
			      if (stream.url === source) {
				      option.setAttribute('selected', 'selected');
			      }
			      select.appendChild(option);
		      }
		      select.disabled = false;
		      select.addEventListener('change', function () {
			      element.src = this.value;
		      });
	      } else {
		      select.style.display = 'none';
	      }
	      return source;
      });
      player.load()
      await this.delay(500)
      if (gotFlag)
	      console.log("success")
      else
	      console.log("fail")
}
```
delay:

```
delay(delay) {
    return new Promise(resolve => {
        setTimeout(() => {
	        resolve(2);
        }, delay);
    });
},
```

**Comment by anast3t** [comment]
To avoid 403 you need referer field in request header. So you need to send request from server (eg localhost). 
But now the problem is that google fixed something and request returns constant 404

**Comment by grcmichael** [comment]
My videos don't load since yesterday and I noticed the same issue on my console : 
`GET https://images21-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https%3A%2F%2Fwww.youtube.com%2Fget_video_info%3Fvideo_id%3DYizPn42SO6o%26el%3Dembedded%26hl%3Den_US 404` :(

**Comment by p13rnd** [comment]
Guess YT updated due to Copyright and such ...

**Comment by grcmichael** [comment]
So we can't use this lib anymore ?

**Comment by p13rnd** [comment]
> So we can't use this lib anymore ?

There could be some fix, maybe. However using YT Content in a Player Other than YT Player surely isn't intended by YT and maybe against YT TOS as well. If you rely on it working stable I suggest to use the js api.

**Comment by anast3t** [comment]
This CAN be fixed because there are still websites from which you can download (like savefrom.net, on main page you can find nonYT player)

**Comment by anast3t** [comment]
I found what you can use instead of this lib for now: ytdl-core. It returns exactly same data-streaming urls. Maybe i'll try to fix this lib with the source code of that :)

example of usage (nuxt):
![image](https://user-images.githubusercontent.com/26357306/118966924-4f336a00-b994-11eb-9d9b-9ee4a8e91d6e.png)
![image](https://user-images.githubusercontent.com/26357306/118966860-39be4000-b994-11eb-99c2-778502b20d1e.png)
![image](https://user-images.githubusercontent.com/26357306/118967086-78ec9100-b994-11eb-8de4-c94a6307f52d.png)

**Comment by thelevicole** [comment]
I've just published version 4.0.1 which contains a quick fix for this issue.

Instead of making responses directly to the YouTube/Google endpoints, I have setup a middleware API (`https://yt2html5.com/`) to handle the retrieval of video information. It's currently on a shared server so please do not expect rapid responses.

Over the next couple of weeks I will be testing and improving the library and API so keep an eye out for future releases.

Let me know how you get on.
